third_party/protobuf/bazel/analysis/BUILD.bazel

```diff
@@ -1,15 +1,31 @@
-load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
+load("@rules_go//go:def.bzl", "go_library")
+load("@rules_go//proto:def.bzl", "go_proto_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
 load("//rules:copy_proto_output.bzl", "copy_proto_output")
 
+proto_library(
+    name = "analysis_proto",
+    srcs = ["analysis_v2.proto"],
+    visibility = ["//visibility:public"],
+    deps = ["//third_party/protobuf/bazel/build:blaze_query_proto"],
+)
+
 go_proto_library(
-    name = "analysis",
+    name = "analysis_go_proto",
     importpath = "github.com/bazel-contrib/target-determinator/third_party/protobuf/bazel/analysis",
-    proto = "@bazel_tools//src/main/protobuf:analysis_v2_proto",
+    proto = ":analysis_proto",
```

Comment:
What triggered this change? It looks like `@bazel_tools//src/main/protobuf:analysis_v2_proto` is still around in bazel 8.1.0 and it ensures that the protobuf stays in sync with the version of bazel being used, without having to hardcode the protos in the repo. It also means that users using TD as a library might end up with protos that are out of sync with the version of bazel they're using.

Reply:
It now pulls in a stardoc proto, which we don't have a dependency for, and adding proves tricky, though I'd be happy to pair on it. In any case, this change in the underlying proto will mean that users will have to be using a specific version of Bazel to make things work if they're compiling the code themselves. Fortunately, Google are pretty good at schema migration, so pinning to an older version of the proto is a safe thing to do.